src/Features/LanguageServer/Protocol/RequestDispatcher.cs

```diff
@@ -73,16 +73,17 @@ public RequestDispatcher(ImmutableArray<Lazy<AbstractRequestHandlerProvider, IRe
             Contract.ThrowIfNull(handlerEntry, string.Format("Request handler entry not found for method {0}", methodName));
 
             var mutatesSolutionState = handlerEntry.Metadata.MutatesSolutionState;
+            var skipBuildingLSPSolution = handlerEntry.Metadata.SkipBuildingLSPSolution;
```

Comment:
this eems welrd.  instead of asking of this of the metadata.  just ask this of the handler.  It seems like it's just a boolean property you can query from the instance.  

Note: being metadata would make sense if we needed to know this prior to even having an instance.  But that doesn't seem to be hte pattern here.  Thoughts?

Reply:
Yeah, the same is true of `MutatesSolutionState`, and I made the same suggestion about that one on the recent refactor to providers. We can discuss offline with @dibarbet

Reply:
Yeah - I didn't do it at the time because I wasn't 100% certain if we wanted a handler to support multiple requests per instance.  But I think it's reasonable to just limit it to that and add these two as properties to the IRequestHandler interface